RFCs/006-DynamicData-Attribute.md

```diff
@@ -0,0 +1,66 @@
+# RFC 006- DynamicData Attribute for Data Driven Tests
+
+## Summary
+This details the MSTest V2 framework attribute "DynamicData" for data driven tests where test data, declared as properties or methods, can be shared between more than one test cases.
+
+## Motivation
+Often times, the inline test data need to be shared between tests. There could be more than one test cases in different files which can use same test data. In most of these situations, user end up declaring same inline data for multiple test cases using "DataRow" attribute. Test framework should provide feature so that test data, declared as property, can be
+* Easily re-used
+* Easily organized
+
+## Detailed Design
+
+### Requirements
+1. Inline test data can be reused by multiple test cases.
+
+### Proposed solution
+Here is a solution that meets the above requirements:
+
+The property storing test data should be declared as static.
```

Comment:
Some more thoughts:
I think `static` constraint may be limiting. Hypothetical scenario: say user is writing a test with one base test class and two inherited child classes testing two implementations. With static data, user can't override the datasource in the inherited children. There can't be a single test method with a single data source.

Is there an implementation challenge with non-static data source?